marine_benchmark.c

```diff
@@ -77,26 +77,28 @@ int load_cap(char *file, packet **packets, char errbuff[PCAP_ERRBUF_SIZE]) {
 void benchmark(packet packets[], int packet_len, char *bpf, char *display_filter, char *fields[], int fields_len) {
     char err_msg[512];
     int filter_id = marine_add_filter(bpf, display_filter, fields, fields_len, err_msg);
+    struct timespec start_time, end_time;
+
 
     if (filter_id < 0) {
-        fprintf(stderr,"Error creating filter id: %s\n", err_msg);
+        fprintf(stderr, "Error creating filter id: %s\n", err_msg);
         return;
     }
 
     size_t memory_start = get_current_rss();
-    clock_t start = clock();
+    clock_gettime(CLOCK_MONOTONIC_RAW, &start_time);
     for (int i = 0; i < packet_len; ++i) {
         packet p = packets[i];
         marine_result *result = marine_dissect_packet(filter_id, (char *) p.data, p.header->len);
         assert(result->result == 1);
         marine_free(result);
     }
-    clock_t end = clock();
+    clock_gettime(CLOCK_MONOTONIC_RAW, &end_time);
     size_t memory_end = get_current_rss();
 
-    float total_time = ((float) end - start) / CLOCKS_PER_SEC;
-    float pps = (float) packet_len / total_time;
-    float memory_usage = ((float) memory_end - memory_start) / 1024 / 1024;
+    double total_time = (end_time.tv_sec - start_time.tv_sec) + ((end_time.tv_nsec - start_time.tv_nsec) * 1e-9);
+    double pps = packet_len / ((double) total_time);
```

Comment:
`total_time` is already double, why are you casting it?

Reply:
idk